### ISSUE TITLE
Sell Quantum Keystones to Remnant at a better price?

### ISSUE THREAD

**Issue by Tadrix** [endless-sky/endless-sky#3579]
Seems unfair that the only time player gets to sell Hai Keystones to Remnant as Key Stones is that one encounter with the Remnant merchant. (I understand that currently it isn't possible for the outfit to have prices variable by outfitter.)

- One way to go about this could be invisible missions changing player's Keystones to Key Stones upon landing on Remnant world. Admittedly it could become a problem because any other trade then would become irrelevant.
- Another way could be invisible missions of the Unfettered Jump Drive kind, which would activate only after getting the Remnant license. Upon landing, if the player has considerably more keystones than ships (if there is a way to check it, otherwise it could default to 50, or less), the invisible missions prompts if player would like to sell his stones to the outfitter (through the familiar manager or his colleagues on another planets). But, since the Remnant are a little greedy and they sense that the player is willing to trade, they start offering lower prices for the Keystones. Twice lower, 60k apiece (or 3M for 50).
- Or there could be several missions for a fixed amount of keystones, each activating after the previous, and gradually dropping prices for them. Simplest form of it would be a mission (which becomes available when the player gets Remnant license) which repeats twice and exchanges 50 keystones for 4M. After the second time the third mission on infinite repeat activates. and that last one exchanges 50 keystones for 3M. _You have to understand, our market has experienced an influx of key stones, so the prices are dropping,_

**Comment by czartrak** [comment]
But aren't Hai Keystones completely outfits completely? Unless something's wrong with my game, they should be that way even after the remnant merchant

**Comment by KiLEdEnNis** [comment]
@czartrak what are you trying to say? Tadrix proposed more missions where the player can buy Hai Keystones & sell them to the Remnant (who will sell them as Key Stones), nothing else will change

**Comment by tehhowch** [comment]
Seems like these missions can be written without any game changes needed. So, all that is needed is someone to write them.
 - A license appears as a condition, so checking for it in the `to offer` will work
 - Make them `minor`, and omit `job` and `landing` (i.e. the player checks the spaceport for them)
 - To enforce a time delay between them, several options exist:
   - Set a condition in the `on complete`, and also trigger an event after the desired number of days, in which the condition is erased. Have the lack of that condition a prerequisite of the `to offer` for the mission. The next mission in the series would then reference the previous mission's `"mission: done" X` count.
   - Just use separate missions (i.e. no `repeat`), and in the `on complete` of each, use the standard unique timer event that you check for in the next mission's `to offer`. This may be desirable anyway, as the dialogue in each mission can then be different each time.

**Comment by czartrak** [comment]
@KiLEdEnNis what i'm saying is, can't you just go sell hai keystones to the remnant without missions. it seems to me the only purpose missions like these would serve is lore

**Comment by KiLEdEnNis** [comment]
buying outfits & selling them for a high price in other parts of space? I remember something like that getting a no, and I don't remember regular jobs being capable of this, so why not add some repeatable mission like the Unfettered Jump Drive one

**Comment by Zitchas** [comment]
Yes it will. I have been awaiting the opportunity to make a PR to close this issue.

**Comment by Hurleveur** [comment]
Yeah would be nice to get the functionality merged at some point, especially for pirate gameplay, but also for this and things like broken jump drive price, or jump drive price for uhai, more expensive weapons/ships at local places without it messing up the AI etc etc
https://github.com/endless-sky/endless-sky/pull/6404
(also I added so many tests for it that other for UI issues we're mostly safe from bugs, I'll add more tests if bugs are discovered)

**Comment by Zitchas** [comment]
Yep, still waiting. Still looking forward to the mechanics being merged so they can be put to use.
